### PR TITLE
[cifuzz] Add basic end-to-end tests.

### PR DESCRIPTION
**Pull request by jonathanmetzman**
This test doesn't really do diffing or affected fuzzers properly.
Nor does it check the workspace for existence of certain things, nor
does check the filesystem for proof things happened. It is still
a WIP.
Related: #6100

### CODE REVIEW

File: infra/cifuzz/cifuzz_end_to_end_test.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# limitations under the License.
+"""End-to-End tests for CIFuzz."""
+import os
+import tempfile
+import unittest
+
+import run_cifuzz
+
+CIFUZZ_DIR = os.path.dirname(os.path.abspath(__file__))
+EXTERNAL_PROJECT_PATH = os.path.join('CIFUZZ_DIR', 'external-project')
```
Comment (inferno-chromium) [inline infra/cifuzz/cifuzz_end_to_end_test.py:22]: CIFUZZ_DIR instead of 'CIFUZZ_DIR'

File: infra/cifuzz/cifuzz_end_to_end_test.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  """End-to-End tests for CIFuzz."""
+
+  def test_simple(self):
+    """Simple end-to-end test using run_cifuzz.main()."""
+    os.environ['REPOSITORY'] = 'external-project'
+    os.environ['PROJECT_SRC_PATH'] = EXTERNAL_PROJECT_PATH
+    os.environ['BUILD_INTEGRATION_PATH'] = 'cifuzz-build-integration'
+    with tempfile.TemporaryDirectory() as temp_dir:
+      os.environ['WORKSPACE'] = temp_dir
+      # This test is weird because running the fuzzer results in a crash which
```
Comment (inferno-chromium) [inline infra/cifuzz/cifuzz_end_to_end_test.py:37]: Why keep the weirdness, shouldnt it return different error code for this.

**Review by inferno-chromium — APPROVED** [review]

**Review by oliverchang — APPROVED** [review]
When we flesh out the actual tests, can we do this in a way that lets us share the test code with something that can be used to test a real github actions/other CI run?

File: infra/cifuzz/cifuzz_end_to_end_test.py (hunk 1 above)
Comment (jonathanmetzman) [reply infra/cifuzz/cifuzz_end_to_end_test.py]: Done.

File: infra/cifuzz/cifuzz_end_to_end_test.py (hunk 2 above)
Comment (jonathanmetzman) [reply infra/cifuzz/cifuzz_end_to_end_test.py:42]: Done.

**Comment by jonathanmetzman** [comment]
> When we flesh out the actual tests, can we do this in a way that lets us share the test code with something that can be used to test a real github actions/other CI run?

I'm not sure I understand. Is the question, "can this code help with testing that cifuzz works on a real CI system?" The answer is sort of. run_cifuzz will be used by real CI systems so this will test most of CIFuzz on those systems. I don't think this test will help with testing CIFuzz on github actions though because it uses lower-level APIs than this script.

**Pull request merged**